File: LeetCode/LeetCode Practice Problems/1880- Check if Word Equals Summation of Two Words.cpp
class Solution {
public:
    bool isSumEqual(string firstWord, string secondWord, string targetWord) {
        int first = 0 , second = 0;
        int target = 0;
        
        for(char c: firstWord) {
            first = first*10 + (c - 'a');
        }
        
        for(char c: secondWord) {
            second = second*10 + (c - 'a');
        }
        
        for(char c: targetWord) {
            target = target*10 + (c - 'a');
        }
        
        return first + second == target;
        
    }
};

File: LeetCode/LeetCode Practice Problems/1880-Check if Word Equals Summation of Two Words.java
class Solution {
    
    public int findSum(String word) {
        int s = 0;
        for(char c: word.toCharArray()) {
            s = s*10 + (c - 'a');
        }
        return s;
    }
    
    public boolean isSumEqual(String firstWord, String secondWord, String targetWord) {
        int num1 = findSum(firstWord);
        int num2 = findSum(secondWord);
        int sum = findSum(targetWord);
        
        return num1 + num2 == sum;
    }
}
